File: snippets/go/receive_payment.go
package example

import (
	"log"

	"github.com/breez/breez-sdk-go/breez_sdk"
)

func ReceivePayment() error {
	// ANCHOR: receive-payment
	receivePaymentRequest := breez_sdk.ReceivePaymentRequest{
		AmountMsat:  uint64(3_000_000),
		Description: "Invoice for 3000 sats",
	}
	receivePaymentResponse, err := sdk.ReceivePayment(receivePaymentRequest)
	if err != nil {
		return err
	}
	log.Printf("Invoice: %#v", receivePaymentResponse.LnInvoice)
	// ANCHOR_END: receive-payment
	return nil
}

File: snippets/python/src/receive_payment.py
import breez_sdk


def receive_payment(sdk_services):
    try:
        # ANCHOR: receive-payment
        req = breez_sdk.ReceivePaymentRequest(
            amount_msat=3000000,
            description="Invoice for 3 000 sats")
        receive_payment_response = sdk_services.receive_payment(req)

        invoice = receive_payment_response.ln_invoice
        # ANCHOR_END: receive-payment
        return receive_payment_response
    except Exception as error:
        print(error)
        raise